lib/Parse/ParseStmt.cpp

```diff
@@ -1861,8 +1862,14 @@ ParserStatus Parser::parseStmtCondition(StmtCondition &Condition,
   // a variety of common errors situations (including migrating from Swift 2
   // syntax).
   while (true) {
-    Status |= parseStmtConditionElement(result, DefaultID, ParentKind,
-                                        BindingKindStr);
+
+    if (Context.LangOpts.hasFeature(Feature::TrailingComma)) {
```

Comment:
If I’m not mistaken, will this production allow

```swift
guard else { return }
```

 To parse?

Reply:
I think I handled this in the swift-syntax implementation and forget here. I will take I look in both.